src/_nebari/stages/terraform_state/__init__.py

```diff
@@ -56,6 +57,30 @@ def _validate_resource_group_name(cls, value):
 
         return value
 
+    @pydantic.validator("tags")
+    def _validate_tags(cls, tags):
+        max_name_length = 512
+        max_value_length = 256
+        invalid_chars = "<>%&\\?/"
+
+        for tag_name, tag_value in tags.items():
+            if any(char in tag_name for char in invalid_chars):
+                raise ValueError(
+                    f"Tag name '{tag_name}' contains invalid characters. Invalid characters are: `{invalid_chars}`"
+                )
+
+            if len(tag_name) > max_name_length:
+                raise ValueError(
+                    f"Tag name '{tag_name}' exceeds maximum length of {max_name_length} characters."
+                )
+
+            if len(tag_value) > max_value_length:
+                raise ValueError(
+                    f"Tag value '{tag_value}' for tag '{tag_name}' exceeds maximum length of {max_value_length} characters."
+                )
+
+        return tags
+
```

Comment:
Why do we have to define this function twice?  Can't we import in one place or the other?